di/src/main/java/com/interface21/beans/factory/support/DefaultListableBeanFactory.java

```diff
@@ -1,36 +1,98 @@
 package com.interface21.beans.factory.support;
 
+import com.interface21.beans.BeanInstantiationException;
 import com.interface21.beans.factory.BeanFactory;
-import com.interface21.beans.factory.config.BeanDefinition;
+import com.interface21.context.stereotype.Component;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.reflections.Reflections;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
 
 public class DefaultListableBeanFactory implements BeanFactory {
 
-    private static final Logger log = LoggerFactory.getLogger(DefaultListableBeanFactory.class);
+  private static final Logger log = LoggerFactory.getLogger(DefaultListableBeanFactory.class);
+  private static final String STEREOTYPE_PACKAGE = "com.interface21.context.stereotype";
 
-    private final Map<String, BeanDefinition> beanDefinitionMap = new HashMap<>();
+  private final Map<Class<?>, Object> singletonObjects = new HashMap<>();
+  private final String[] basePackages;
 
-    private final Map<Class<?>, Object> singletonObjects = new HashMap<>();
+  public DefaultListableBeanFactory(String... basePackages) {
+    this.basePackages =  basePackages;
+  }
 
-    @Override
-    public Set<Class<?>> getBeanClasses() {
-        return Set.of();
-    }
+  @Override
+  public Set<Class<?>> getBeanClasses() {
+    return singletonObjects.keySet();
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T> T getBean(final Class<T> clazz) {
+    return (T) singletonObjects.get(clazz);
```

Comment:
```suggestion
    Object bean = singletonObjects.get(clazz);
    return clazz.cast(bean);
```

이러면 warning도 사라질 것 같네요!